modules/twinklefluff.js

```diff
@@ -45,40 +45,61 @@ Twinkle.fluff = {
-		// is used to show rollback/vandalism links for IP ranges
-		if( mw.config.get('wgCanonicalSpecialPageName') === "Contributions" && (mw.config.exists('wgRelevantUserName') || !!$('#sp-contributions-footer-anon-range')[0])) {
-			//Get the username these contributions are for
-			var username = mw.config.get('wgRelevantUserName');
-			if( Twinkle.getPref('showRollbackLinks').indexOf('contribs') !== -1 ||
-				( mw.config.get('wgUserName') !== username && Twinkle.getPref('showRollbackLinks').indexOf('others') !== -1 ) ||
-				( mw.config.get('wgUserName') === username && Twinkle.getPref('showRollbackLinks').indexOf('mine') !== -1 ) ) {
-				var list = $("#mw-content-text").find("ul li:has(span.mw-uctop)");
-
-				var revNode = document.createElement('strong');
-				var revLink = Twinkle.fluff.buildLink('SteelBlue', 'rollback');
-				revNode.appendChild(revLink);
-
-				var revVandNode = document.createElement('strong');
-				var revVandLink = Twinkle.fluff.buildLink('Red', 'vandalism');
-				revVandNode.appendChild(revVandLink);
-
-				list.each(function(key, current) {
-					var href = $(current).find(".mw-changeslist-diff").attr("href");
-					if (href) {
-						current.appendChild( document.createTextNode(' ') );
-						var tmpNode = revNode.cloneNode( true );
-						tmpNode.firstChild.setAttribute( 'href', href + '&' + Morebits.queryString.create( { 'twinklerevert': 'norm' } ) );
-						current.appendChild( tmpNode );
-						current.appendChild( document.createTextNode(' ') );
-						tmpNode = revVandNode.cloneNode( true );
-						tmpNode.firstChild.setAttribute( 'href', href + '&' + Morebits.queryString.create( { 'twinklerevert': 'vand' } ) );
-						current.appendChild( tmpNode );
-					}
-				});
+	contributions: {
+		findPages: function() {
+			// $('sp-contributions-footer-anon-range') relies on the fmbox
+			// id in [[MediaWiki:Sp-contributions-footer-anon-range]] and
+			// is used to show rollback/vandalism links for IP ranges
+			if( mw.config.exists('wgRelevantUserName') || !!$('#sp-contributions-footer-anon-range')[0]) {
+				//Get the username these contributions are for
+				var username = mw.config.get('wgRelevantUserName');
+				if( Twinkle.getPref('showRollbackLinks').indexOf('contribs') !== -1 ||
+					( mw.config.get('wgUserName') !== username && Twinkle.getPref('showRollbackLinks').indexOf('others') !== -1 ) ||
+					( mw.config.get('wgUserName') === username && Twinkle.getPref('showRollbackLinks').indexOf('mine') !== -1 ) ) {
+					var list = $("#mw-content-text").find("ul li:has(span.mw-uctop):has(.mw-changeslist-diff)");
+					var titles = $(list).find(".mw-changeslist-date").map(function(){return $(this).attr("title");}).get();
+
+					var query = {
+						'action': 'query',
+						'prop': 'info',
+						'intestactions': 'edit',
+						'titles': titles.join('|').replace(/ /g,'_')
+					};
+
+					var wikipedia_api = new Morebits.wiki.api('Checking for prior AfDs', query, Twinkle.fluff.contributions.main);
```

Comment:
That is not at all what this API request is doing

Reply:
Ahahaha good catch.  Will fix when I get home.